### ISSUE TITLE
Inconsistent count between Azure Cosmos DB and Emulator

### ISSUE THREAD

**Issue by refractivSteve** [Azure/azure-cosmos-dotnet-v3#2244]
Apologies if this is not the right place to log this bug.

**Describe the bug**
When performing a count query with an order by on documents which may have an undefined property, I get different counts between the emulator (v2.11.10.0) and that in the cloud.  

**To Reproduce**
Add the following two documents into both Azure Cosmos DB and the emulator :

`{"id": "missing-property-1",
    "ResponseItemsData": {
        "always-available": "Foo"
    }}`
    
`{"id": "missing-property-2",
    "ResponseItemsData": {
        "always-available": "Foo",
        "sometimes-missing": "Bar"
    }}`
    
Then run the following query:
```
SELECT VALUE COUNT(1) 
FROM root 
ORDER BY root["ResponseItemsData"]["sometimes-missing"] ASC
```

**Expected behavior**
I either expect to get a count of 2 in both instances or a count of 1 in both

**Actual behavior**
I get a count of 2 in the emulator and a count of 1 in the cloud

**Environment summary**
SDK Version: 3.16.0
OS Version: Window

Removing the order by clause removes the inconsistency, but I thought it best to create awareness of the issue.

**Comment by j82w** [comment]
@timsander1 or @sboshra or @bchong95 was there a fix done to the service to cause this difference? If so can you please work on getting a new emulator released that contains the changes.

**Comment by timsander1** [comment]
Hi @refractivSteve. What is the indexing policy of each container? If you've used a /* vs explicitly including paths, the results for ORDER BY with missing items may be different. Learn more: https://docs.microsoft.com/en-us/azure/cosmos-db/sql-query-order-by#documents-with-missing-fields

**Comment by refractivSteve** [comment]
Hi @timsander1.  The index policies on both are identical, as per the below. The NextPollDueAt is unrelated to this ticket, but thought it best to copy the policy verbatim.

The paths aren't explicitly set as the ResponseItemsData object properties are schema-less and defined as part of user input (in the real application).  I have already structured the application code in a way that deals with missing item order bys as per the link you posted. 

After I pushed the application to Azure I altered the code to remove the order clauses from the count query to fix the issue. 

```
{
    "indexingMode": "consistent",
    "automatic": true,
    "includedPaths": [
        {
            "path": "/*"
        },
        {
            "path": "/NextPollDueAt/?"
        }
    ],
    "excludedPaths": [
        {
            "path": "/\"_etag\"/?"
        }
    ]
}
```
